crates/primitives-traits/src/node.rs

```diff
@@ -77,46 +78,74 @@ impl NodePrimitives for () {
 }
 
 /// Helper trait that sets trait bounds on [`NodePrimitives`].
-pub trait FullNodePrimitives
-where
-    Self: NodePrimitives<
-            Block: FullBlock<Header = Self::BlockHeader, Body = Self::BlockBody>,
-            BlockHeader: FullBlockHeader,
-            BlockBody: FullBlockBody<Transaction = Self::SignedTx>,
-            SignedTx: FullSignedTx,
-            TxType: FullTxType,
-            Receipt: FullReceipt,
-        > + Send
-        + Sync
-        + Unpin
-        + Clone
-        + Default
-        + fmt::Debug
-        + PartialEq
-        + Eq
-        + 'static,
+pub trait FullNodePrimitives:
+    NodePrimitives<
+    Block: FullBlock<Header = Self::BlockHeader, Body = Self::BlockBody> + 'static,
+    BlockHeader: FullBlockHeader + 'static,
+    BlockBody: FullBlockBody<Transaction = Self::SignedTx> + 'static,
+    SignedTx: FullSignedTx + 'static,
+    TxType: FullTxType + 'static,
+    Receipt: FullReceipt + 'static,
+>
 {
 }
 
 impl<T> FullNodePrimitives for T where
     T: NodePrimitives<
-            Block: FullBlock<Header = Self::BlockHeader, Body = Self::BlockBody>,
-            BlockHeader: FullBlockHeader,
-            BlockBody: FullBlockBody<Transaction = Self::SignedTx>,
-            SignedTx: FullSignedTx,
-            TxType: FullTxType,
-            Receipt: FullReceipt,
-        > + Send
-        + Sync
-        + Unpin
-        + Clone
-        + Default
-        + fmt::Debug
-        + PartialEq
-        + Eq
-        + 'static
+        Block: FullBlock<Header = Self::BlockHeader, Body = Self::BlockBody> + 'static,
+        BlockHeader: FullBlockHeader + 'static,
+        BlockBody: FullBlockBody<Transaction = Self::SignedTx> + 'static,
+        SignedTx: FullSignedTx + 'static,
+        TxType: FullTxType + 'static,
+        Receipt: FullReceipt + 'static,
+    >
 {
 }
 
 /// Helper adapter type for accessing [`NodePrimitives`] receipt type.
 pub type ReceiptTy<N> = <N as NodePrimitives>::Receipt;
+
+/// Helper trait that links [`NodePrimitives::Block`] to [`NodePrimitives::BlockHeader`],
+/// [`NodePrimitives::BlockBody`] and [`NodePrimitives::SignedTx`].
+pub trait BlockPrimitives:
```

Comment:
unclear what the benefit here is
what does this unblock?

Reply:
it links the block type to the header, body and signed tx types. it should be used in places where `FullNodePrimitives` is not needed, and simplifies having to specify the super trait of `BlockPrimitives` everywhere, i.e. `N: NodePrimitives<Block: Block<Header = N::BlockHeader, ..>, ..>` to link the blockparts and tx to the block type